modules/daggerverse/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"context"
+	"dagger/daggerverse/internal/dagger"
+	"fmt"
+	"time"
+
+	"github.com/google/go-github/v66/github"
+)
+
+type Daggerverse struct{}
+
+// Deploy preview environment running Dagger main: dagger call deploy-preview-with-dagger-main --github-token=env:GITHUB_PAT
+func (h *Daggerverse) DeployPreviewWithDaggerMain(
+	ctx context.Context,
+	githubToken *dagger.Secret,
+) error {
+	today := time.Now()
+	date := today.Format("2006-01-02")
+
+	repo := "github.com/dagger/dagger.io"
+	// clone dagger.io - private repository, requires a github token
+	gh := dag.Gh(dagger.GhOpts{
+		Token: githubToken,
+		Repo:  repo,
+	})
+
+	// make a change so that a new Daggerverse deployment will be created
+	daggerio := gh.Repo().Clone(
```

Comment:
is there a reason to not use the `dag.Git` function here?

Reply:
The only reason was the `Token: githubToken` support which made `gh` the easiest choice.

What would the `dag.Git` alternative look like?

Reply:
there's a `dag.Git().WithAuthToken()` which should also work.

Reply:
OK, let me try that and see how it goes.

Reply:
I could be using it wrong. Given the following:

```go
	today := time.Now()
	date := today.Format("2006-01-02")

	// clone dagger.io - private repository, requires a github token
	// make a change so that a new Daggerverse deployment will be created
	repo := "github.com/dagger/dagger.io"
	daggerio := dag.Git(repo).WithAuthToken(githubToken).Branch("main").Tree().
		WithNewFile("daggerverse/CREATE_PREVIEW_ENVIRONMENT", today.String())

	branch := fmt.Sprintf("test-e2e-integration-%s-with-dagger-main", date)

	// open a PR so that it creates a new Daggerverse preview environment running Dagger main
	gh := dag.Gh(dagger.GhOpts{
		Token: githubToken,
		Repo:  repo,
	})
	err = gh.WithSource(daggerio).
		WithGitExec([]string{"checkout", "-b", branch}).
		WithGitExec([]string{"add", "daggerverse/CREATE_PREVIEW_ENVIRONMENT"}).
		WithGitExec([]string{"config", "user.email", *emails[0].Email}).
		WithGitExec([]string{"config", "user.name", *user.Name}).
		WithGitExec([]string{"commit", "-am", fmt.Sprintf("'Test Daggerverse with Dagger Engine main version @ %s'", date)}).
		WithGitExec([]string{"push", "--force", "origin", branch}).
		PullRequest().Create(
		ctx,
		dagger.GhPullRequestCreateOpts{
			Assignees: []string{*user.Login},
			Fill:      true,
			Labels:    []string{"preview", "area/daggerverse"},
			Head:      branch,
		},
	)

	return err
```

I'm getting the following error:

```console
  ✘ GhPullRequest.create(assignees: ["gerhard"], fill: true, head: "test-e2e-integration-2024-11-06-with-dagger-main", labels: ["preview", "area/daggerverse"]): Void 1.3s
  ! call function "Create": process "/runtime" did not complete successfully: exit code: 2
  ┃ invoke: input: container.import.withEnvVariable.withEnvVariable.withExec.withEnvVariable.withSecretVariable.withEnvVariable.withWorkdir.withMountedDirectory.withExec.sync res
  ┃ : process "gh pr create --assignee gerhard --fill --head test-e2e-integration-2024-11-06-with-dagger-main --label preview --label area/daggerverse" did not complete successfu
  ┃  exit code: 1
  ┃
  ┃ Stderr:
  ┃ could not compute title or body defaults: failed to run git: fatal: ambiguous argument 'origin/main...test-e2e-integration-2024-11-06-with-dagger-main': unknown revision or p
  ┃ not in the working tree.
  ┃ Use '--' to separate paths from revisions, like this:
  ┃ 'git <command> [<revision>...] -- [<file>...]'
    ✘ Container.withExec(args: ["gh", "pr", "create", "--assignee", "gerhard", "--fill", "--head", "test-e2e-integration-2024-11-06-with-dagger-main", "--label", "preview", "--label", "area/daggerverse"]): Container! 0.6s
    ┃ could not compute title or body defaults: failed to run git: fatal: ambiguous argument 'origin/main...test-e2e-integration-2024-11-06-with-dagger-main': unknown revision or
    ┃ not in the working tree.
    ┃ Use '--' to separate paths from revisions, like this:
    ┃ 'git <command> [<revision>...] -- [<file>...]'
    ✘ Container.sync: ContainerID! 0.8s
    ! process "gh pr create --assignee gerhard --fill --head test-e2e-integration-2024-11-06-with-dagger-main --label preview --label area/daggerverse" did not complete successfully: exit code: 1

Error: response from query: input: daggerverse.deployPreviewWithDaggerMain resolve: call function "DeployPreviewWithDaggerMain": process "/runtime" did not complete successfully: exit code: 2

Stdout:
invoke: input: gh.withSource.withGitExec.withGitExec.withGitExec.withGitExec.withGitExec.withGitExec.pullRequest.create resolve: call function "Create": process "/runtime" did not complete successfully: exit code: 2

Stdout:
invoke: input: container.import.withEnvVariable.withEnvVariable.withExec.withEnvVariable.withSecretVariable.withEnvVariable.withWorkdir.withMountedDirectory.withExec.sync resolve: process "gh pr create --assignee gerhard --fill --head test-e2e-integration-2024-11-06-with-dagger-main --label preview --label area/daggerverse" did not complete successfully: exit code: 1

Stderr:
could not compute title or body defaults: failed to run git: fatal: ambiguous argument 'origin/main...test-e2e-integration-2024-11-06-with-dagger-main': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

Since I still need `gh` to create the actual PR, I don't see any advantage to using  advantage to using `dag.Git()` in this context. Leaving this as is for now.

Reply:
We can always improve this in a follow-up. Module is working as expected, will merge & move on now 💪